web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -456,7 +456,11 @@ private Promise<Void> authUpdate() {
                     // token is no longer valid, signal deauth for re-login
                     // TODO deephaven-core#2564 fire an event for the UI to re-auth
                     checkStatus(status);
-                    reject.onInvoke(status.getDetails());
+                    if (status.getDetails() == null || status.getDetails().isEmpty()) {
+                        reject.onInvoke(status.getDetails());
```

Comment:
Should this be the opposite? Reject with `status.getDetails()` if it's not null or empty.

Reply:
Yes, it definitely should. I must have somehow not rebuilt when I un-reverted this...

Reply:
Fixed, tested by running the IDE with PSK enabled on the latest server, to confirm I didn't get the 16 error code, but the proper message `Authentication details invalid`.